src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -496,182 +1249,315 @@ bool emberAfThermostatClusterSetActivePresetRequestCallback(
+    // previous StartPresetsSchedulesEditRequest, return UNSUPPORTED_ACCESS.
+    if (gThermostatAttrAccess.GetOriginatorScopedNodeId(endpoint) != sourceNodeId)
+    {
+        commandObj->AddStatus(commandPath, imcode::UnsupportedAccess);
+        return true;
+    }
+
+    Delegate * delegate = GetDelegate(endpoint);
+
+    if (delegate == nullptr)
+    {
+        ChipLogError(Zcl, "Delegate is null");
+        return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::InvalidInState);
+    }
+
+    // Clear the timer, discard the changes and set PresetsEditable to false.
+    return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::Success);
 }
 
 bool emberAfThermostatClusterCommitPresetsSchedulesRequestCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::Thermostat::Commands::CommitPresetsSchedulesRequest::DecodableType & commandData)
 {
-    // TODO
-    return false;
-}
+    EndpointId endpoint = commandPath.mEndpointId;
+    Delegate * delegate = GetDelegate(endpoint);
 
-bool emberAfThermostatClusterSetTemperatureSetpointHoldPolicyCallback(
-    chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-    const chip::app::Clusters::Thermostat::Commands::SetTemperatureSetpointHoldPolicy::DecodableType & commandData)
-{
-    // TODO
-    return false;
-}
+    if (delegate == nullptr)
+    {
+        ChipLogError(Zcl, "Delegate is null");
+        return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::InvalidInState);
+    }
 
-int16_t EnforceHeatingSetpointLimits(int16_t HeatingSetpoint, EndpointId endpoint)
-{
-    // Optional Mfg supplied limits
-    int16_t AbsMinHeatSetpointLimit = kDefaultAbsMinHeatSetpointLimit;
-    int16_t AbsMaxHeatSetpointLimit = kDefaultAbsMaxHeatSetpointLimit;
+    // If presets are not editable, return INVALID_IN_STATE.
+    if (!gThermostatAttrAccess.GetPresetsEditable(endpoint))
+    {
+        return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::InvalidInState);
+    }
 
-    // Optional User supplied limits
-    int16_t MinHeatSetpointLimit = kDefaultMinHeatSetpointLimit;
-    int16_t MaxHeatSetpointLimit = kDefaultMaxHeatSetpointLimit;
+    ScopedNodeId sourceNodeId = GetSourceScopedNodeId(commandObj);
 
-    // Attempt to read the setpoint limits
-    // Absmin/max are manufacturer limits
-    // min/max are user imposed min/max
+    // If the node id sending the CommitPresetsSchedulesRequest command is not the same as the one which send the
+    // StartPresetsSchedulesEditRequest, return UNSUPPORTED_ACCESS.
+    if (gThermostatAttrAccess.GetOriginatorScopedNodeId(endpoint) != sourceNodeId)
+    {
+        return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::UnsupportedAccess);
```

Comment:
Won't this clear the pending presets, though?  Why are we allowing some random other node to do that?

Reply:
we shouldn't will fix.

Reply:
fixed.